src/cpp/core/system/PosixShellUtils.cpp

```diff
@@ -23,9 +23,8 @@ namespace shell_utils {
 
 std::string escape(const std::string& arg)
 {
-   using namespace boost;
-   regex pattern("[\\\\$`!\\n\"]", regex_constants::normal);
-   return "\"" + regex_replace(arg, pattern, "\\$1") + "\"";
+   boost::regex pattern("(\\\\|\\$|`|!|\n|\")");
```

Comment:
We might actually need to replace newlines explicitly with the characters `\n` here as well (since I'm not sure if placing a `\` in front a literal newline character will have the intended effect)